File: pkg/storage/json/visa.go
package json

type Visa struct {
	Passport    string `json:"passport"`
	Destination string `json:"destination"`
	Duration    string `json:"dur"`
	Status      string `json:"text"`
	LastUpdated string `json:"last_updated"`
	Error       Error  `json:"error"`
}

type Error struct {
	Status bool   `json:"status"`
	Error  string `json:"error"`
}

type VisaCountryInfo struct {
	VF  []string `json:"VF"`
	VOA []string `json:"VOA"`
	VR  []string `json:"VR"`
	CB  []string `json:"CB"`
	NA  []string `json:"NA"`
}

File: lib/models/visa.dart
import 'package:equatable/equatable.dart';
// ignore: depend_on_referenced_packages
import 'package:json_annotation/json_annotation.dart';
import 'error.dart';

part 'visa.g.dart';

@JsonSerializable()
class Visa extends Equatable {
  final String passport, destination, dur, status, category, lastUpdated;
  final Error error;

  Visa({
    required this.passport,
    required this.destination,
    required this.dur,
    required this.status,
    required this.category,
    required this.lastUpdated,
    required this.error,
  });

  factory Visa.hasError(String message) => Visa(
        passport: '',
        destination: '',
        dur: '',
        status: '',
        category: '',
        lastUpdated: DateTime.now().toIso8601String(),
        error: Error(error: message, status: true),
      );

  factory Visa.fromJson(Map<String, dynamic> json) => _$VisaFromJson(json);

  Map<String, dynamic> toJson() => _$VisaToJson(this);

  @override
  List<Object?> get props =>
      [passport, destination, dur, status, category, lastUpdated, error];
}
